src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -156,6 +167,43 @@ public IActionResult DrainStatus([FromServices] IScriptHostManager scriptHostMan
             }
         }
 
+        [HttpPost]
+        [Route("admin/host/resume")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> Resume([FromServices] IScriptHostManager scriptHostManager)
+        {
+            var currentState = scriptHostManager.State;
+
+            _logger.LogDebug($"Received request to resume a draining host - host status: {currentState.ToString()}");
+
+            if (currentState != ScriptHostState.Running
+                || !Utility.TryGetHostService(scriptHostManager, out IDrainModeManager drainModeManager))
+            {
+                _logger.LogDebug("The host is not in a state where we can resume.");
+                return StatusCode(StatusCodes.Status409Conflict);
+            }
+
+            try
+            {
+                await _drainModeSemaphore.WaitAsync();
```

Comment:
Should the state check be moved inside of this semaphore?

Reply:
Ah whoops, was supposed to do that! Thanks for pointing out :D